pkg/utils/utils_virtual.go

```diff
@@ -42,9 +44,12 @@ var (
 )
 
 const (
-	ospMetaDataDir = "/host/var/config/openstack/2018-08-27"
-	ospNetworkData = ospMetaDataDir + "/network_data.json"
-	ospMetaData    = ospMetaDataDir + "/meta_data.json"
+	ospMetaDataDir     = "/host/var/config/openstack/2018-08-27"
+	ospMetaDataBaseUrl = "http://169.254.169.254/openstack/latest"
```

Comment:
please use `2018-08-27` instead of `latest` for consistency and safety against a stable API.